source/manual/kubernetes-infrastructure.html.md

```diff
@@ -7,93 +7,175 @@ parent: "/manual.html"
+You can fetch the logs for all running containers in a pod. Take one of the pods from the output above:
 
-- `k config set-context --current --namespace=apps`
+```sh
+k logs account-api-568d46d6f7-4pnld
+```
 
-#### Testing changes to a rake task.
+Or you can fetch the logs using its 'app deployment' name. This will choose one of the Account API pods at random, much like the old jumpbox methods:
 
-For special router publisher as an example:
+```sh
+k logs deploy/account-api
+```
 
-1. Deploy your branch (see above)
-2. Run the following command(s):
+## Open a rails console
+
+Open a rails console with:
 
+```sh
+k exec -it deploy/whitehall-admin -- rails c
 ```
-BASE_PATH=whatever
 
-kubectl run -n apps --image 172025368201.dkr.ecr.eu-west-1.amazonaws.com/special-route-publisher special-route-publisher -- rake "publish_one_special_route[${BASE_PATH?}]"
+If you try doing this having assumed a `readonly` role when setting context, you'll see a permissions error:
 
-kubectl logs -fnapps special-route-publisher
+```sh
+Defaulted container "app" out of: app, nginx
+Error from server (Forbidden): pods "whitehall-admin-6bb47c48d-wkkxk" is forbidden: User "christopher.ashton-user" cannot create resource "pods/exec" in API group "" in the namespace "apps"
+```
 
-kubectl -napps delete po/special-route-publisher
+You need write permissions to be able to open a console (`poweruser` or above).
+
+## Open a shell
+
+Open a shell (requires `poweruser` or above):
 
+```sh
+k exec -it deploy/whitehall-admin -- bash
 ```
 
-There’s also a [command in the PR](https://github.com/alphagov/special-route-publisher/pull/257) that might work.
+Note that the prompt will have prefix that may feel a little unfamiliar: `I have no name!@whitehall-admin-6bb47c48d-wkkxk:/app$`.
 
-## Dashboards and alerting
+If you try to open a shell on Router, it will fail:
 
-- The detailed Sidekiq dashboard is not yet available for EKS (the [old one](https://sidekiq-monitoring.integration.govuk.digital/publishing-api/queues "‌") still looks at EC2) but you can see queue lengths and age-of-oldest-job on the [Sidekiq Grafana dashboard](https://grafana.eks.production.govuk.digital/d/2Yy8PzmVk "‌").
-- Dashboards are on [https://grafana.eks.production.govuk.digital/](https://grafana.eks.production.govuk.digital/?orgId=1 "‌") and no longer require VPN. Don’t get confused by the [old Grafana](https://grafana.production.govuk.digital/ "‌"), which will show mostly empty graphs about the old EC2 setup.
-- Most Icinga alerts are irrelevant now, except for:
-    - “scheduled publications in Whitehall not queued” / “overdue publications in Whitehall”
-    - “Travel Advice email alert check” / “Medical Safety Email alert check”
-    - Anything to do with crawler/mirrors
-    - Anything to do with Licensify
+```sh
+$ k exec -it deploy/router -- bash                                                               
```

Comment:
It'd be more helpful to the reader to mention the way that works first! :)

Including this bit somewhere is still a good idea though since it makes the error message searchable.